pkg/medusa/reconcile.go

```diff
@@ -143,8 +143,8 @@ func UpdateMedusaInitContainer(dcConfig *cassandra.DatacenterConfig, medusaSpec
 	}
 	setImage(medusaSpec.ContainerImage, restoreContainer)
 	restoreContainer.SecurityContext = medusaSpec.SecurityContext
-	restoreContainer.Env = medusaEnvVars(medusaSpec, dcConfig, logger, "RESTORE")
-	restoreContainer.VolumeMounts = medusaVolumeMounts(medusaSpec, dcConfig, logger)
+	restoreContainer.Env = medusaEnvVars(medusaSpec, clusterName, logger, "RESTORE")
```

Comment:
For my understanding:

Originally, passing in the entire dcConfig value, now scoped to clusterName only.  What types of values were being used from dcConfig for Medusa?

Reply:
In these parts of the code, only `dcConfig.Cluster` was being used. Previously it was the name of the K8ssandraCluster object that was passed as cluster name for the DC, but now it can be overriden. We still want the objects here to be named after the K8ssandraCluster name and not the Cassandra cluster name, thus the change (dcConfig doesn't contain any reference to the original K8ssandraCluster name if it's overridden).